modules/labs-react/radio/lib/RadioInput.tsx

```diff
@@ -0,0 +1,187 @@
+      transition: 'border 200ms ease, background 200ms',
+      opacity: disabled && variant === 'inverse' ? '.4' : '1',
+    },
+
+    '&:checked:after': {
+      backgroundColor: variant === 'inverse' ? themePrimary.main : colors.frenchVanilla100,
+      border: `5px solid`,
+      borderColor: variant === 'inverse' ? colors.frenchVanilla100 : themePrimary.main,
+    },
+
+    '&:focus:after, &:focus:hover:after': {
+      outline: 'transparent',
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 2,
+        separation: 0,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    '&:focus:checked:after, &:focus:hover:checked:after': {
+      outline: 'transparent',
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 2,
+        separation: 2,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    ...mouseFocusBehavior({
+      '&:focus:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+      '&:focus:hover:after, &:focus:active:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+      '&:focus:hover:checked:after, &:focus:active:checked:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+    }),
+  })
+);
+
+const RadioInputWrapper = styled(Flex)<Pick<RadioLabelProps, 'disabled' | 'variant'>>(
+  {
+    // Hover Ripple element
+    '::before': {
+      content: "''",
+      borderRadius: borderRadius.circle,
+      height: radioHeight,
+      transition: 'box-shadow 150ms ease-out',
+      width: radioWidth,
+      pointerEvents: 'none',
+    },
+    '&:hover:before': {
+      boxShadow: `0 0 0 ${rippleRadius}px ${colors.soap200}`,
+    },
+  },
+  ({variant}) => ({
+    '::before': {
+      opacity: variant === 'inverse' ? '0.4' : '1',
+    },
+  })
+);
+
+const useRadioInput = createElemPropsHook(useRadioModel)(
+  (model, ref, elemProps: {value?: string} = {}) => {
+    const {disabled, variant} = React.useContext(RadioLabelContext);
+    return {
+      disabled: disabled,
+      variant: variant,
+      checked:
+        elemProps.value === model.state.value || elemProps.value === model.state.initialValue,
+      'aria-checked':
+        elemProps.value === model.state.value || elemProps.value === model.state.initialValue,
+      onChange(event: React.ChangeEvent<HTMLInputElement>) {
```

Comment:
Another area looking for feedback. I can pass `checked` to the input manually and it will style it correctly, but `aria-checked` is still false since it's looking for model value and elemProps value to match.